lib/events.js

```diff
@@ -307,3 +345,10 @@ function spliceOne(list, index) {
     list[i] = list[k];
   list.pop();
 }
+
+function arrayClone(arr, len) {
+  var arr_clone = new Array(len);
+  for (var i = 0; i < len; i += 1)
```

Comment:
@micnic right, that's why i switch to `slice()` when the array has >= 50 elements. from my testing that seems to be about the time when a manual copy starts getting slower.